File: src/main/webapp/js/controller/adminUserController.js
//控制层
adminApp.controller('adminUserController',function ($scope,$location,adminUserService) {

    $scope.user = "";

    $scope.oldPassword = "";
    $scope.newPassword = "";
    $scope.focusoldPassword = false;
    $scope.oldPasswordFlag = false;

    $scope.getUser = function () {
        adminUserService.getLoginUser().success(function (response) {
            $scope.user = response.queryResult.map.user;
        });
    };
    $scope.savePassword = function () {
        if($scope.oldPassword != "" && $scope.oldPasswordFlag == false && $scope.newPassword !=""){
            console.log($scope.oldPassword + $scope.newPassword);
            adminUserService.changePassword($scope.user,$scope.newPassword).success(function (response) {
                console.log(response);
                alert(response.message);
            });
        }else{
            alert("请确认原始密码正确");
        }
        $scope.oldPassword = "";
        $scope.newPassword = "";
    };

    $scope.checkOldPassword = function () {
        adminUserService.checkOldPassword($scope.user,$scope.oldPassword ).success(function (response) {
            $scope.oldPasswordFlag = !response.queryResult.map.oldPasswordFlag;
        });
    }
});

File: src/main/java/com/kfgs/admin/controller/AdminUserController.java
package com.kfgs.admin.controller;

import com.kfgs.admin.service.AdminUserService;
import com.kfgs.domain.response.QueryResponseResult;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.web.bind.annotation.*;

/**
 * Date: 2020-05-07-10-27
 * Module:
 * Description:
 *
 * @author:
 */
@RestController
@RequestMapping("admin/userController")
public class AdminUserController {

    @Autowired
    AdminUserService adminUserService;

    @GetMapping("/getLoginUser")
    public QueryResponseResult getLoginUser(){
       return adminUserService.getLoginUser();
    }

    @PostMapping("/checkOldPassword")
    public QueryResponseResult checkOldPassword(@RequestParam("username") String username, @RequestParam("oldPassword")String oldPassword){
        return adminUserService.checkOldPassword(username,oldPassword);
    }

    @PostMapping("/changePassword")
    public QueryResponseResult changePassword(@RequestParam("username") String username, @RequestParam("newPassword")String newPassword){
        return adminUserService.changePassword(username,newPassword);
    }
}
